File: 07_Power of two/Ideal Solutions/powerOfTwo.java
import java.io.*;
import java.lang.*;
import java.util.*;


class Main{
	//This function finds out if the given number is a power of two or not.
	public static boolean powerOfTwo(int n){
		return (n>0) && ((n&n-1)==0);
	}
	//Main Function
	public static void main(String[] args){
		int n = 4;
		if (powerOfTwo(n)) System.out.println("Yes");
		else{
			System.out.println("No");
		} 
	}
}

File: 07_Power of two/Ideal Solutions/powerOfTwo.js
/* 
Given a positive integer N. 
Check if it is a power of two or not.
*/

const powerOfTwo = (N) => {
  while (N % 2 == 0) {
    N /= 2;
  }
  return N === 1 ? true : false;
};
console.log(powerOfTwo(18));
